File: BackEnd/Zalo_mini/Zalo_mini/Models/conversation.cs
using System;
using System.Collections.Generic;

namespace Zalo_mini.Models
{
    public partial class conversation
    {
        public conversation()
        {
            calls = new HashSet<call>();
            conversation_participants = new HashSet<conversation_participant>();
            conversation_tag_assignments = new HashSet<conversation_tag_assignment>();
            messages = new HashSet<message>();
        }

        public long id { get; set; }
        public string conversation_type { get; set; } = null!;
        public long? group_id { get; set; }
        public string? name { get; set; }
        public string? avatar_url { get; set; }
        public long? last_message_id { get; set; }
        public DateTime? last_message_at { get; set; }
        public bool? is_active { get; set; } = true;
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }

        public virtual chat_group? group { get; set; }
        public virtual message? last_message { get; set; }
        public virtual ICollection<call> calls { get; set; }
        public virtual ICollection<conversation_participant> conversation_participants { get; set; }
        public virtual ICollection<conversation_tag_assignment> conversation_tag_assignments { get; set; }
        public virtual ICollection<message> messages { get; set; }
        public virtual ICollection<conversation_deleted> conversation_deleted { get; set; } = new HashSet<conversation_deleted>();
    }
}

File: BackEnd/java/ZaloMini/src/main/java/com/devmam/zalomini/entities/Conversation.java
package com.devmam.zalomini.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.Instant;
import java.util.Date;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity(name = "Conversation")
@Table(name = "conversations")
public class Conversation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Lob
    @Column(name = "conversation_type", nullable = false)
    private String conversationType;

    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "group_id")
    private ChatGroup group;

    @Column(name = "name")
    private String name;

    @Lob
    @Column(name = "avatar_url")
    private String avatarUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "last_message_id")
    private Message lastMessage;

    @Column(name = "last_message_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date lastMessageAt;

    @ColumnDefault("1")
    @Column(name = "is_active")
    private Boolean isActive;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "created_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "updated_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date updatedAt;

}